hack/run-wmcb-ci-e2e-test.sh

```diff
@@ -3,57 +3,78 @@ set -o errexit
-    \? )
-      echo "Usage: $0 [-v] [-s]"
-      exit 0
-      ;;
-  esac
-done
-
-# TODO: Add steps to execute code that will kick off the e2e tests in CI
+# downloads the oc binary. We can use oc directly from /tmp
+get_oc(){
+  # Download the oc binary only if it is not already available
+  # We do not validate the version of oc if it is available already
+  if type oc >/dev/null 2>&1; then
+    which oc
+    return
+  fi
+
+  OC_DIR=/tmp/oc
+  curl -L -s https://mirror.openshift.com/pub/openshift-v4/clients/ocp/4.5.8/openshift-client-linux-4.5.8.tar.gz -o openshift-origin-client-tools.tar.gz \
+    && tar -xzf openshift-origin-client-tools.tar.gz \
+    && mv oc $OC_DIR \
+    && rm -rf ./openshift*
+
+  echo $OC_DIR
+}
+
+# delete_deployment deletes the deployed test pod
+delete_deployment(){
+  if ! $1 delete -f internal/test/wmcb/deploy/deploy.yaml -n default; then
+    echo "no deployment found"
+  fi
+}
+
 WMCO_ROOT=$(dirname "${BASH_SOURCE}")/..
-WMCB_TEST_DIR=$WMCO_ROOT/internal/test/wmcb
-
-# Create a temporary directory for the repos
-TMP_DIR=$(mktemp -d)
-trap "rm -rf $TMP_DIR" EXIT
-
-pushd "$TMP_DIR"
-# Build hybrid-overlay-node.exe
-$GIT_CLONE https://github.com/openshift/ovn-kubernetes.git
-cd ovn-kubernetes/go-controller/
-make windows
-HYBRID_OVERLAY_BIN=$TMP_DIR"/ovn-kubernetes/go-controller/_output/go/bin/windows/hybrid-overlay-node.exe"
-
-# Build kubelet.exe
-cd "$TMP_DIR"
-$GIT_CLONE https://github.com/openshift/kubernetes.git
-cd kubernetes
-KUBE_BUILD_PLATFORMS=windows/amd64 make WHAT=cmd/kubelet
-KUBELET_BIN=$TMP_DIR"/kubernetes/_output/local/bin/windows/amd64/kubelet.exe"
-popd
-
-# Build the unit test binary
 cd "${WMCO_ROOT}"
-make build-wmcb-unit-test
-make build-wmcb-e2e-test
 
+# If ARTIFACT_DIR is not set, create a temp directory for artifacts
+ARTIFACT_DIR=${ARTIFACT_DIR:-}
+if [ -z "$ARTIFACT_DIR" ]; then
+  ARTIFACT_DIR=`mktemp -d`
+  echo "ARTIFACT_DIR is not set. Artifacts will be stored in: $ARTIFACT_DIR"
+  export ARTIFACT_DIR=$ARTIFACT_DIR
+fi
+
+OC=$(get_oc)
+
+if ! $OC create secret generic cloud-private-key --from-file=private-key.pem=$KUBE_SSH_KEY_PATH -n default; then
+    echo "cloud-private-key already present"
+fi
+if ! $OC create secret generic aws-creds --from-file=credentials=$AWS_SHARED_CREDENTIALS_FILE -n default; then
+    echo "aws credentials already present"
+fi
+if ! $OC apply -f internal/test/wmcb/deploy/role.yaml -n default; then
+    echo "role already present"
+fi
+if ! $OC apply -f internal/test/wmcb/deploy/rolebinding.yaml -n default; then
+    echo "rolebinding already present"
+fi
+
+sed -i "s~ARTIFACT_DIR_VALUE~${ARTIFACT_DIR}~g" internal/test/wmcb/deploy/deploy.yaml
+sed -i "s~REPLACE_IMAGE~${WMCB_IMAGE}~g" internal/test/wmcb/deploy/deploy.yaml
+
+# deploy the test pod on test cluster
+if ! $OC apply -f internal/test/wmcb/deploy/deploy.yaml -n default; then
+    echo "application already deployed"
+fi
+
+# Sleep for 1 minute while the deployment is being created.
+sleep 1m
```

Comment:
Please wait for the deployment to be ready rather than a sleep.

Reply:
replaced the 1 minute sleep by a while loop which waits for the pod to enter running state and starts streaming logs once it enters a running state.